### ISSUE TITLE
Conda prompt should ignore base

### ISSUE THREAD

**Issue by spencermathews** [conda/conda#8169]
## Current Behavior

Using the new recommended method of enabling conda in the shell (4.4.0+, `conda activate`) even the base environment is displayed on the prompt. So now, my everyday prompt is prefixed with `(base)`, and this is just clutter.

The conda configuration options seem poorly equipped to handle this scenario. Under the previous (`source activate`) approach the `changeps1` option was sufficient since the base env was only added to `PATH`. Since it was not actually activated the prompt was not affected.

The simplest solution might be to treat the base env as an exception in terms of affecting the prompt. An additional configuration option could be added to allow it to be shown if desired.

### Steps to Reproduce

Using BASH, instead of exporting `PATH` in `.bashrc` do the following:

```
. $HOME/anaconda3/etc/profile.d/conda.sh
conda activate base
```

## Expected Behavior

The base environment, which is active by default in the common use case, should not be show in the shell prompt. This is the historical behavior resulting from the legacy instructions for enabling conda in a shell, and for consistency it should remain this way.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

(abbreviated)
```
     active environment : base
            shell level : 1
          conda version : 4.5.11
    conda-build version : not installed
         python version : 3.7.0.final.0
               platform : linux-64
```
</p></details>

**Comment by msarahan** [comment]
While I agree with your sentiment here, this will not be of sufficient priority for the team to actually work on for a very long time.  PRs to fix this would be welcomed, if you get annoyed enough and find time.

**Comment by rongou** [comment]
This is what the latest installer put into my `.bashrc`:
```bash
# added by Miniconda3 4.5.12 installer
# >>> conda init >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$(CONDA_REPORT_ERRORS=false '/home/***/miniconda3/bin/conda' shell.bash hook 2> /dev/null)"
if [ $? -eq 0 ]; then
    \eval "$__conda_setup"
else
    if [ -f "/home/***/miniconda3/etc/profile.d/conda.sh" ]; then
        . "/home/***/miniconda3/etc/profile.d/conda.sh"
        CONDA_CHANGEPS1=false conda activate base
    else
        \export PATH="/home/rou/miniconda3/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda init <<<
```
and I get `(base)` in my prompt as well. Seems like a simple fix is to put `CONDA_CHANGEPS1=false` in front of the `\eval` line. Does that sound like the right fix?

**Comment by kalefranz** [comment]
`CONDA_CHANGEPS1=false conda activate base` is in effect a single-shot disabling of `(base) `.  

Conda 4.6 introduced a new `env_prompt` config parameter:

```
$ conda config --describe env_prompt
# # env_prompt (str)
# #   Template for prompt modification based on the active environment.
# #   Currently supported template variables are '{prefix}', '{name}', and
# #   '{default_env}'. '{prefix}' is the absolute path to the active
# #   environment. '{name}' is the basename of the active environment
# #   prefix. '{default_env}' holds the value of '{name}' if the active
# #   environment is a conda named environment ('-n' flag), or otherwise
# #   holds the value of '{prefix}'. Templating uses python's str.format()
# #   method.
# #
# env_prompt: '({default_env}) '
```

It's pretty customizable (you can even use ANSI color codes with the right escaping).  I'd look at logic for how that's used along with the `changeps1` config parameter to implement a more permanent solution than the single-shot option above.  Ideally it could be done without adding yet another new boolean config parameter.  The parameter list is pretty long at this point... starting to get into config parameter paralysis territory.

**Comment by rongou** [comment]
Right, my fix just removes `(base)` when you open a new shell, it's not a permanent fix.

Is it possible to do conditional formatting with `env_prompt`?

**Comment by spencermathews** [comment]
Thanks all for the suggestions.

While not ideal, the single-shot approach seems sufficient as a stopgap. At least the base env is not indicated on every newly spawned shell.

The `env_prompt` config parameter seems promising. One simple approach might be to add an option which behaved like `{default_env}` but ignored base.

I noticed #7781 also mentions the base prompt.

Unfortunately docs are not particularly up to date regarding the newer approaches to configuring the shell, and are not at all helpful regarding the `__conda_setup` strategy.

**Comment by mark-heiple** [comment]
I am running macOS Sierra and Anaconda3, the docs are inconsistent with errors returned when trying to use conda activate.

As installed, the conda command is not found. The docs say to add this to my .bash_profile (I installed Anaconda3 for all users, so it is in /Anaconda3

export PATH="/anaconda3/bin:$PATH"

After doing so, conda commands such as "conda info" work, but conda activate does not, it said to do this:

 $ echo ". /anaconda3/etc/profile.d/conda.sh" >> ~/.bash_profile

or for all users do this:

$ sudo ln -s /anaconda3/etc/profile.d/conda.sh /etc/profile.d/conda.sh

The 2nd option didn't work, so I went with the first.

However, it also says this:

Previous to conda 4.4, the recommended way to activate conda was to modify PATH in
your ~/.bash_profile file.  You should manually remove the line that looks like

    export PATH="/anaconda3/bin:$PATH"

**^^^ The above line should NO LONGER be in your ~/.bash_profile file! ^^^**

However, without that in the PATH, it can't find the conda command so nothing works. This is inconsistent with the docs here: https://docs.anaconda.com/anaconda/install/mac-os/

**The installer prompts “Do you wish the installer to prepend the Anaconda install location to PATH in your /home/<user>/.bash_profile ?” We recommend “yes”.

NOTE: If you enter “no”, specify the path to Anaconda.

Add export PATH="/<path to anaconda>/bin:$PATH" in .bashrc or .bash_profile.
Or set the PATH variable: export PATH="/<path to anaconda>/bin:$PATH"
Replace “<path to anaconda>” with the actual path to your Anaconda installation.**

What are the consequences of having it in the path? The main thing I noticed is the Anaconda python runs instead of the system python by default.

What is the correct method for configuring conda to work. Is the default system python intended to still work? If so, how do I get it to work?

**Comment by iabraham** [comment]
What about running: `conda config --set auto_activate_base false` as pointed out [here](https://stackoverflow.com/questions/54429210/how-do-i-prevent-conda-from-activating-the-base-environment-by-default) ?

**Comment by spencermathews** [comment]
@iabraham the issue is not about preventing the base env from being activated. Assuming a typical install, à la `conda init`, wherein conda becomes the default python, it's only about preventing `(base)` from being prefixed to the prompt of every new shell. A full solution would also prevent the prefixing when returning to base (`conda deactivate`) from some other environment.

**Comment by prisae** [comment]
:+1: for getting a solution on that. If you use conda as your default environment, you definitely do not want your everyday terminal have cluttered with `(base)`, it simply does not make sense.

**Comment by mingwandroid** [comment]
Running conda as your system is pretty radical!

**Comment by prisae** [comment]
Not as my system, but as my default environment. Basically what always was standard by setting
```
export PATH="$HOME/anaconda3/bin:$PATH"
```
in `.bashrc`, which was even automated by conda. I, and I think many others, use/run conda like this all the time.

But since this change, it means that EACH terminal starts with `(base) `, which is pretty much useless; even more so for tasks that are not Python related.

**Comment by mingwandroid** [comment]
Anaconda Distribution isn't just python. Far from it.

**Comment by prisae** [comment]
Fair enough. Still, it does not change the point/problem. The anaconda install always recommended at the end of the install-script to add the path at the end of `.bashrc`, so I think a lot of users got used to it, in fact, I would not want to be without it. However, suddenly from one release to another modifying every standard terminal prompt is not nice. Which is reflected by the many different issues that cropped up here. I just wanted to support this feeling which many other expressed in this and other threads.

**Comment by msarahan** [comment]
I'm not opposed to having a way to change this, but I can't dedicate our resources to make that change right now.  We have too many other issues that have more impact, such as conda's speed and new features that we need to add for our enterprise customers.

If anyone is feeling up for a PR to add a setting for this, I'll be happy to review and merge it.

I understand why some people don't like the prompt addition, since it is nominally a change in behavior.  I also think it's nice to have it there, because "explicit is better than implicit."  Yes, things change over time, and there will always be people who like a change and people who do not.  Showing this in the prompt will remain our default, but it's fine with me if people want to add a setting that hides the base env addition.

**Comment by prisae** [comment]
I can absolutely understand this, and it is fine. I just wanted to express my support for the intent of this issue (and similar ones).

**Comment by brandonsueur** [comment]
Hi,

I confirm, for me there is only the request below that works!

```bash
$ conda config --set auto_activate_base false
```

Thank you all for your feedback 🙂

**Comment by zpincus** [comment]
A terminal window is often ~80 characters wide by default. Adding `(base)` to every prompt eats almost 10% of every line on a standard terminal. This is far from a case where one can casually say "explicit is better than implicit" without considering other costs. Especially because having conda active-by-default is the usual configuration, showing `(base)` is almost pure noise.

This was a fairly intrusive change to (a) foist on all users in their daily terminal environment, and then (b) disavow any responsibility for by claiming that the dev team is "too busy". Obviously they weren't too busy to implement this misfeature in the first place!

**Comment by Keou0007** [comment]
@zpincus hits the nail on the head. Lazy selfish devs make changes people don't like and then refuse to fix it or take responsibility.
For now I have used the simple option of `changeps1: false` to get rid of the pointless clutter you've added. But now the PS1 doesn't change when I activate a new environment, I just have to remember which env I have active. 
So what you've actually achieved here is removing a useful feature that I and many others liked using. Well done to you.

**Comment by zpincus** [comment]
Probably best to not get too abusive here (I wouldn't go so far as to say "lazy selfish devs", considering that conda is open source and the devs are giving us their work for free). For the record, here's what I've done to fix this problem, as per [this StackOverflow post](https://stackoverflow.com/questions/55171696/how-to-remove-base-from-terminal-prompt-after-updating-conda/55172508):
```
CONDA_ROOT=~/miniconda3
mkdir -p "${CONDA_ROOT}/etc/conda/activate.d"
cat > "${CONDA_ROOT}/etc/conda/activate.d/remove_base_ps1.sh" << EOF
PS1="$(echo $PS1 | sed 's/(base) //')"
EOF
```

Basically, it uses a script hook that the devs did provide to fix `PS1` to remove `(base) ` if present. A little hackish, but not the worst thing in the world at all.

**Comment by msarahan** [comment]
Conda is open source, and PRs are welcome.  Snark and insults are not.
